### PR TITLE
feat: Course outline - Section Publish

### PR DESCRIPTION
**Pull request by vladislavkeblysh**
- Add publish modal, api, tests
- Update section card, card header tests

[2u-354](https://youtrack.raccoongang.com/issue/2U-354)

### CODE REVIEW

File: src/course-outline/card-header/CardHeader.jsx [hunk 1]
```diff
@@ -65,18 +69,24 @@ const CardHeader = ({
           )}
         </Button>
       </OverlayTrigger>
-      <Dropdown data-testid="section-card-header__menu" className="ml-auto">
+      <Dropdown data-testid="section-card-header__menu" className="ml-auto" onClick={onMenuButtonClick}>
```
Comment (PKulkoRaccoonGang) [inline src/course-outline/card-header/CardHeader.jsx:72]: Looks better

```suggestion
      <Dropdown 
          data-testid="section-card-header__menu" 
          className="ml-auto" 
          onClick={onMenuButtonClick}
      >
```

File: src/course-outline/section-card/SectionCard.jsx [hunk 2]
```diff
@@ -2,10 +2,18 @@ import React, { useState } from 'react';
 import PropTypes from 'prop-types';
 import { Badge, Button } from '@edx/paragon';
 
+import { useDispatch } from 'react-redux';
 import CardHeader from '../card-header/CardHeader';
```
Comment (PKulkoRaccoonGang) [inline src/course-outline/section-card/SectionCard.jsx:6]:
```suggestion
import { Badge, Button } from '@edx/paragon';
import { useDispatch } from 'react-redux';

import CardHeader from '../card-header/CardHeader';
```

File: src/course-outline/card-header/CardHeader.jsx (hunk 1 above)
Comment (vladislavkeblysh) [reply src/course-outline/card-header/CardHeader.jsx]: Done

File: src/course-outline/section-card/SectionCard.jsx (hunk 2 above)
Comment (vladislavkeblysh) [reply src/course-outline/section-card/SectionCard.jsx]: Done

File: src/course-outline/data/thunk.js [hunk 3]
```diff
@@ -150,3 +150,22 @@ export function updateCourseSectionHighlightsQuery(sectionId, highlights) {
     }
   };
 }
+
+export function publishCourseSectionQuery(sectionId) {
+  return async (dispatch) => {
+    dispatch(updateSavingStatus({ status: RequestStatus.IN_PROGRESS }));
```
Comment (ruzniaievdm) [inline src/course-outline/data/thunk.js:156]: please change to 
`dispatch(updateSavingStatus({ status: RequestStatus.PENDING }));` for `POST` method

File: src/course-outline/section-card/SectionCard.jsx [hunk 4]
```diff
@@ -30,13 +38,19 @@ const SectionCard = ({ section, children, onOpenHighlightsModal }) => {
     setIsExpanded((prevState) => !prevState);
   };
 
+  const handleMenuButtonClick = () => {
```
Comment (ruzniaievdm) [inline src/course-outline/section-card/SectionCard.jsx:41]: please change to `handleClickMenuButton`

**Review by ruzniaievdm — CHANGES_REQUESTED** [review]

File: src/course-outline/section-card/SectionCard.jsx (hunk 4 above)
Comment (vladislavkeblysh) [reply src/course-outline/section-card/SectionCard.jsx]: Done

File: src/course-outline/data/thunk.js (hunk 3 above)
Comment (vladislavkeblysh) [reply src/course-outline/data/thunk.js]: Done

File: src/course-outline/card-header/CardHeader.jsx [hunk 5]
```diff
@@ -22,11 +22,15 @@ const CardHeader = ({
   title,
   sectionStatus,
   isExpanded,
-  handleExpand,
+  onExpand,
+  onPublishClick,
+  onMenuButtonClick,
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/course-outline/card-header/CardHeader.jsx]: Done

File: src/course-outline/card-header/CardHeader.jsx [hunk 6]
```diff
@@ -22,11 +22,15 @@ const CardHeader = ({
   title,
   sectionStatus,
   isExpanded,
-  handleExpand,
+  onExpand,
+  onPublishClick,
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/course-outline/card-header/CardHeader.jsx]: Done

File: src/course-outline/hooks.jsx [hunk 7]
```diff
@@ -84,6 +86,16 @@ const useCourseOutline = ({ courseId }) => {
     closeHighlightsModal();
   };
 
+  const handlePublishModalOpen = () => {
+    openPublishModal();
+  };
+
+  const handlePublishSectionSubmit = () => {
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/course-outline/hooks.jsx]: Done

**Review by ruzniaievdm — APPROVED** [review]

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Review by wowkalucky — APPROVED** [review]
OK.

**Pull request merged**